File: Embarcadero/XE5/CPP/InteractiveGestures/TapAndHold/TapHoldForm.cpp
//---------------------------------------------------------------------------

#include <fmx.h>
#include <System.UITypes.hpp>
#pragma hdrstop

#include "TapHoldForm.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.fmx"
TTapHold *TapHold;
//---------------------------------------------------------------------------
__fastcall TTapHold::TTapHold(TComponent* Owner)
	: TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TTapHold::FormGesture(TObject *Sender, const TGestureEventInfo &EventInfo,
          bool &Handled)
{
	// if a long tap gesture is detected
	if (EventInfo.GestureID == igiLongTap) {
		// Show a message
		Title->Text = Title->Text.sprintf(L"LongTap at %d, %d",
			static_cast<int>(EventInfo.Location.X), static_cast<int>(EventInfo.Location.Y));
	}
}
//---------------------------------------------------------------------------

File: Embarcadero/XE8/ObjectPascal/InteractiveGestures/TapAndHold/TapHoldForm.pas
unit TapHoldForm;

interface

uses
  System.SysUtils, System.Types, System.UITypes, System.Classes, System.Variants,
  FMX.Types, FMX.Controls, FMX.Forms, FMX.Dialogs, FMX.Gestures, FMX.Objects,
  FMX.StdCtrls;

type
  TTapHold = class(TForm)
    GestureManager1: TGestureManager;
    Image1: TImage;
    ToolBar1: TToolBar;
    Title: TLabel;
    procedure FormGesture(Sender: TObject; const EventInfo: TGestureEventInfo;
      var Handled: Boolean);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  TapHold: TTapHold;

implementation

{$R *.fmx}
{$R *.LgXhdpiPh.fmx ANDROID}

procedure TTapHold.FormGesture(Sender: TObject;
  const EventInfo: TGestureEventInfo; var Handled: Boolean);
begin
  { if a long tap gesture is detected }
  if EventInfo.GestureID = System.UITypes.igiLongTap then
    { show a message }
    Title.Text := Format('LongTap at %d, %d seen %s',
               [Round(EventInfo.Location.X), Round(EventInfo.Location.Y),
                FormatDateTime('nn:ss',Now)]);
end;

end.
